### PR TITLE
Tweaks to Packaging, Testing, Documentation, and CLI sections

### PR DESCRIPTION
**Pull request by lgorenstein**

### CODE REVIEW

File: docs/source/tutorial/commandline.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@ Command-line Interfaces
 In addition to including routines in a package that yourself and others can import within other
 projects, typically it is a good idea to expose those elements that can stand alone as a command
 line utility (e.g., a high level function whose parameters might map in a straightforward way to
-command-line arguments). Structuring your project so that it can manifest as both a library and a
+command-line arguments). Often such command-line functionality may even be the primary driving force behind initial development of a program (*"I need a script that does X"*), and only later it may evolve into *"I need to incorporate capability X into a higher-level process"*.  Structuring your project so that it can manifest as both a library and a
```
Comment (glentner) [inline docs/source/tutorial/commandline.rst:9]: This is good.

File: docs/source/tutorial/commandline.rst [hunk 2]
```diff
@@ -33,8 +33,8 @@ was created in, all they see is the command-line interface.
 Command-line Arguments
 ^^^^^^^^^^^^^^^^^^^^^^
 
-The distinguishing characteristic of a "real" command-line tool and `merely` a script is the
-presents of what could be a rich set of arguments and options passable to the command.
+The distinguishing characteristic of a "real" command-line tool or `merely` a script is the
+presence of what could be a rich set of arguments and options passable to the command.
```
Comment (glentner) [inline docs/source/tutorial/commandline.rst]: there, their, they're ... 😬

File: docs/source/tutorial/documentation.rst [hunk 3]
```diff
@@ -135,7 +135,7 @@ Here we've followed a particular style guide; Sphinx uses `Google's documentatio
 <https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html>`_ by default to parse
 your docstrings, more on this later! NumPy's `documentation guidelines
 <https://numpydoc.readthedocs.io/en/latest/>`_ are also a great reference for more information
-about what and how to document your code. There are other style guides you might prefer.
+about what and how to document in your code. There are other style guides you might prefer.
```
Comment (glentner) [inline docs/source/tutorial/documentation.rst:138]: I think you're technically correct on this one, though I do only ever hear "document your code".

File: docs/source/tutorial/documentation.rst [hunk 4]
```diff
@@ -313,7 +313,7 @@ and follow some simple setup procedures. Not only will it host your Sphinx docum
 Type Annotations
 ^^^^^^^^^^^^^^^^
 
-A relatively new concept in Python,
+A relatively new concept in Python (3.5+),
```
Comment (glentner) [inline docs/source/tutorial/documentation.rst:316]: Indeed. Might also be worth noting that this is still a growing feature set. Forward references on things like class methods that return an instance of `self` don't work until 3.7+.

File: docs/source/tutorial/documentation.rst [hunk 5]
```diff
@@ -379,8 +379,8 @@ We can add annotations to our code as follows.
 
     From `PEP 484 <https://www.python.org/dev/peps/pep-0484/#id27>`_:
 
-    *Rather than requiring that users write import numbers and then use numbers.Float etc.,
+    *Rather than requiring that users write ``import numbers`` and then use ``numbers.Float`` etc.,
```
Comment (glentner) [inline docs/source/tutorial/documentation.rst:382]: One of the annoying drawbacks of restructured text is that you can't combine styling. So the fact that this is emphasized precludes it from containing verbatim text (like teletype). 😱

File: docs/source/tutorial/packaging.rst [hunk 6]
```diff
@@ -353,6 +353,16 @@ without needing to re-install it every time.
 
     $ pip install -e . --user
 
+As a side note, using pip's ``--user`` functionality might have some unexpected
```
Comment (glentner) [inline docs/source/tutorial/packaging.rst:356]: Nice. I'll add make this into a note I think.

**Review by glentner — APPROVED** [review]

File: docs/source/tutorial/documentation.rst (hunk 3 above)
Comment (lgorenstein) [reply docs/source/tutorial/documentation.rst:138]: To me it's the 'what' that changes it.  "How to document your code", but "what to document _in_ your code" - hence I leaned for the "in" in the combined "what and how" construct.

File: docs/source/tutorial/documentation.rst (hunk 5 above)
Comment (lgorenstein) [reply docs/source/tutorial/documentation.rst:382]: Uggggh, how silly ☹️

**Pull request merged**